src/util.cpp

```diff
@@ -418,6 +418,7 @@ bool SoftSetArg(const std::string& strArg, const std::string& strValue)
     if (mapArgs.count(strArg))
         return false;
     mapArgs[strArg] = strValue;
+    _mapMultiArgs[strArg].push_back(strValue);
```

Comment:
I believe this is also not correct - other threads may be using mapMultiArgs or the vector in it at the same time